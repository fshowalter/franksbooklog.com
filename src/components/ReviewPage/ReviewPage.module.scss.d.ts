export const authorLinkCss: string;
export const containerCss: string;
export const coverCss: string;
export const headerAuthorCss: string;
export const headerContainerCss: string;
export const headerKindCss: string;
export const headerMetaCss: string;
export const headerOriginalTitleCss: string;
export const headerSeparatorCss: string;
export const headerTitleCss: string;
export const headerYearCss: string;
export const hideDesktopCss: string;
export const olderViewingsContainerCss: string;
export const olderViewingsDateCss: string;
export const olderViewingsDateIconCss: string;
export const olderViewingsHeadingCss: string;
export const olderViewingsListCss: string;
export const olderViewingsListItemCss: string;
export const olderViewingSlugCss: string;
export const olderViewingsVenueCss: string;
export const olderViewingsViaCss: string;
export const progressContainerCss: string;
export const progressMilestoneCss: string;
export const readingTimeContainerCss: string;
export const relatedAvaterCss: string;
export const relatedCss: string;
export const relatedHeaderCss: string;
export const relatedHeadingCss: string;
export const relatedMoreCss: string;
export const relatedMoviesSectionCss: string;
export const relatedNameCss: string;
export const reviewContentCss: string;
export const reviewCss: string;
export const reviewDateIconCss: string;
export const reviewGradeCss: string;
export const reviewMetaCss: string;
export const reviewsListCss: string;
export const reviewsListItemCss: string;
export const slugContainerCss: string;
export const slugDateCss: string;
export const slugFootnoteCss: string;
export const slugOnCss: string;
export const slugViaCss: string;
export const termCss: string;
