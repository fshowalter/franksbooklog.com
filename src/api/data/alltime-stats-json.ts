import { promises as fs } from "node:fs";
import { z } from "zod";

import { perfLogger } from "~/utils/performanceLogger";

import { DistributionSchema } from "./DistributionSchema";
import { MostReadAuthorSchema } from "./MostReadAuthorSchema";
import { getContentPath } from "./utils/getContentPath";

const alltimeStatsFile = getContentPath("data", "all-time-stats.json");

/**
 * Extended distribution schema for grade data that includes a sort value.
 * Used specifically for organizing review grades in proper order.
 */
const GradeDistributionSchema = DistributionSchema.extend({
  sortValue: z.number(),
});

/**
 * Zod schema for all-time statistics data loaded from JSON.
 * Validates the structure of comprehensive reading statistics across all years.
 */
const AlltimeStatsJsonSchema = z.object({
  bookCount: z.number(),
  decadeDistribution: z.array(DistributionSchema),
  editionDistribution: z.array(DistributionSchema),
  gradeDistribution: z.array(GradeDistributionSchema),
  kindDistribution: z.array(DistributionSchema),
  mostReadAuthors: z.array(MostReadAuthorSchema),
  reviewCount: z.number(),
  workCount: z.number(),
});

/**
 * Type representing all-time reading statistics data.
 * Contains comprehensive metrics including counts, distributions, and top authors.
 */
export type AlltimeStatsJson = z.infer<typeof AlltimeStatsJsonSchema>;

/**
 * Loads and validates all-time reading statistics from JSON file.
 * Provides comprehensive lifetime reading metrics including book counts,
 * grade distributions, most read authors, and categorical breakdowns.
 *
 * @returns Promise resolving to validated all-time statistics data
 * @throws ZodError if the JSON data doesn't match the expected schema
 *
 * @example
 * ```typescript
 * const stats = await alltimeStatsJson();
 * console.log(`Total books read: ${stats.bookCount}`);
 * console.log(`Top author: ${stats.mostReadAuthors[0].name}`);
 * ```
 */
export async function alltimeStatsJson(): Promise<AlltimeStatsJson> {
  return await perfLogger.measure("alltimeStatsJson", async () => {
    const json = await fs.readFile(alltimeStatsFile, "utf8");
    const data = JSON.parse(json) as unknown[];

    return AlltimeStatsJsonSchema.parse(data);
  });
}
