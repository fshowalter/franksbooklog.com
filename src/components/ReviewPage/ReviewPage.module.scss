@use "./src/styles/breakpoints";
@use "./src/styles/layout";

.containerCss {
  display: grid;
  grid-template-areas: ". header ." ". reviews ." ". related .";
  grid-template-columns:
    minmax(var(--layout-grid-margin), 1fr) minmax(auto, 520px)
    minmax(var(--layout-grid-margin), 1fr);
  grid-template-rows: auto;
  justify-content: space-between;
  margin: 3rem auto 0;

  @media (min-width: breakpoints.$tablet) {
    grid-template-areas: "header header header" ". reviews ." ". related .";
  }

  @media (min-width: breakpoints.$desktop) {
    grid-template-areas: "header" "reviews" "related";
    grid-template-columns: 1fr;
    grid-template-rows: none;
    max-width: unset;
    row-gap: 3rem;
  }

  @media (min-width: breakpoints.$max) {
    margin-top: 0;
  }
}

.reviewGradeCss {
  display: block;
  height: 32px;
  margin: 0 auto;
  width: auto;

  @media (min-width: breakpoints.$desktop) {
    height: 20px;
    margin: 0 1ch 0 0;
  }
}

.reviewsListItemCss {
  position: relative;
}

.reviewCss {
  @media (min-width: breakpoints.$desktop) {
    align-items: start;
    display: grid;
    grid-template-areas: "slug . meta" "review . meta";
    grid-template-columns: minmax(auto, 550px) layout.$spacer-10 1fr;
    grid-template-rows: auto 1fr;
    max-width: unset;
    padding: 0;
    text-align: left;
  }
}

.reviewsListCss {
  grid-area: reviews;
  padding-top: layout.$spacer-4;

  @media (min-width: breakpoints.$desktop) {
    border-top: 1px solid var(--color-border-default);
    padding-top: layout.$spacer-6 - 1;
  }
}

.reviewsListItemCss:not(:first-child) .reviewGradeCss {
  height: 24px;
  margin-top: layout.$spacer-5;

  @media (min-width: breakpoints.$desktop) {
    height: 18px;
    margin: 0 1ch 0 0;
  }
}

.reviewsListItemCss:not(:first-child) {
  padding-top: layout.$spacer-5;
  position: relative;

  @media (min-width: breakpoints.$desktop) {
    padding-top: layout.$spacer-9;
  }
}

.reviewsListItemCss:not(:first-child)::after {
  background-color: var(--color-bg-default);
  content: url("/assets/film-icon.svg");
  display: block;
  height: layout.$spacer-3;
  left: calc(50% - layout.$spacer-4);
  padding: 0 layout.$spacer-3;
  position: absolute;
  top: layout.$spacer-4;
  width: layout.$spacer-7;
  z-index: 100;
}

.reviewsListItemCss:not(:first-child)::before {
  border-top: solid 1px var(--color-border-default);
  content: " ";
  height: 1px;
  left: 25%;
  position: absolute;
  top: calc(layout.$spacer-5 - 1px);
  width: 50%;
}

.reviewDateIconCss {
  display: none;
}

.slugContainerCss {
  font-feature-settings: "tnum";
  font-size: 16px;
  letter-spacing: 0.5px;
  line-height: layout.$spacer-4;
  text-align: center;

  @media (min-width: breakpoints.$desktop) {
    align-items: center;
    display: flex;
    grid-area: slug;
    text-align: left;
  }
}

.slugOnCss {
  color: var(--color-fg-subtle);
  display: block;
  margin: 0 auto;

  @media (min-width: breakpoints.$desktop) {
    display: inline;
  }
}

.slugDateCss {
  color: var(--color-fg-subtle);
  display: block;
  font-weight: 700;
  margin: 0 auto;

  @media (min-width: breakpoints.$desktop) {
    display: inline;
  }
}

.coverCss {
  grid-area: cover;
  justify-self: center;
}

.headerTitleCss {
  font-size: 32px;
  grid-area: title;
  line-height: layout.$spacer-5;

  @media (min-width: breakpoints.$desktop) {
    font-size: 36px;
    line-height: layout.$spacer-6;
  }
}

.headerAuthorCss {
  font-size: 18px;
  grid-area: author;
}

.headerYearCss {
  color: var(--color-fg-subtle);
  font-size: 16px;
  grid-area: year;
}

.headerKindCss {
  color: var(--color-fg-subtle);
  font-size: 14px;
  grid-area: kind;
  letter-spacing: 1px;
  text-transform: uppercase;
}

.headerContainerCss {
  display: grid;
  grid-area: header;
  grid-template-areas: "cover" "title" "author" "kind" "year";
  grid-template-columns: 1fr;
  grid-template-rows: auto;
  justify-items: center;
  max-width: unset;
  row-gap: 1rem;
  text-align: center;
  text-align: left;

  @media (min-width: breakpoints.$tablet) {
    grid-template-areas: ". cover . title ." ". cover . author ." ". cover . kind ." ". cover . year ." ". cover . . .";
    grid-template-columns: minmax(var(--layout-grid-margin), 1fr) 250px layout.$spacer-6 auto minmax(
        var(--layout-grid-margin),
        1fr
      );
    justify-items: unset;
  }

  @media (min-width: breakpoints.$desktop) {
    grid-template-areas: "cover . title" "cover . author" "cover . kind" "cover . year" "cover . .";
    grid-template-columns: 250px layout.$spacer-6 1fr;
    grid-template-rows: auto;
    max-width: unset;
    padding: 0;
    row-gap: 1rem;
    text-align: left;
  }
}

.reviewContentCss {
  margin-top: layout.$spacer-5;

  @media (min-width: breakpoints.$desktop) {
    grid-area: review;
    margin-top: layout.$spacer-3;
    position: relative;
  }
}

.reviewMetaCss {
  background-color: var(--color-bg-subtle);
  grid-area: meta;
  margin: layout.$spacer-7 0;
  padding: layout.$spacer-2 layout.$spacer-6 layout.$spacer-6;

  @media (min-width: breakpoints.$desktop) {
    align-self: flex-start;
    margin: unset;
  }
}

.termCss {
  font-weight: 700;
  line-height: layout.$spacer-4;
  margin-top: layout.$spacer-4;
}

.readingTimeContainerCss {
  font-feature-settings: "tnum";
}

.progressContainerCss {
  line-height: layout.$spacer-4;
  margin-top: layout.$spacer-4;
}

.progressMilestoneCss {
  font-weight: 700;
}

.authorLinkCss {
  color: var(--color-fg-accent);
}
