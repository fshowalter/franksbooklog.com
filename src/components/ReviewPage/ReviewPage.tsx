import { graphql, Link } from "gatsby";
import { GatsbyImage, IGatsbyImageData } from "gatsby-plugin-image";
import React from "react";
import toSentenceArray from "../../utils/to-sentence-array";
import DateIcon from "../DateIcon";
import Grade from "../Grade";
import Layout from "../Layout";
import RenderedMarkdown from "../RenderedMarkdown";
import Seo from "../Seo";
import {
  authorLinkCss,
  containerCss,
  coverCss,
  headerAuthorCss,
  headerContainerCss,
  headerKindCss,
  headerTitleCss,
  headerYearCss,
  progressContainerCss,
  progressMilestoneCss,
  readingTimeContainerCss,
  reviewContentCss,
  reviewCss,
  reviewDateIconCss,
  reviewGradeCss,
  reviewMetaCss,
  reviewsListCss,
  reviewsListItemCss,
  slugContainerCss,
  slugDateCss,
  slugOnCss,
  termCss,
} from "./ReviewPage.module.scss";

function buildStructuredData(pageData: PageQueryResult) {
  const gradeMap: { [index: string]: number } = {
    A: 5,
    B: 4,
    C: 3,
    D: 2,
    F: 1,
  };

  return {
    "@context": "http://schema.org",
    "@type": "AggregateRating",
    itemReviewed: {
      "@type": "Book",
      name: pageData.work.title,
      image: pageData.work.seoImage.childImageSharp.resize.src,
    },
    reviewCount: pageData.work.reviews.length,
    ratingValue: gradeMap[pageData.work.lastReviewGrade[0]],
  };
}

/**
 * Renders a review page.
 */
export default function ReviewPage({
  data,
}: {
  data: PageQueryResult;
}): JSX.Element {
  const structuredData = buildStructuredData(data);
  const { work } = data;
  const authorNames = toSentenceArray(
    work.authors.map((author) => author.name)
  ).join(", ");

  return (
    <Layout>
      <Seo
        pageTitle={`${work.title} by ${authorNames}`}
        description={`A review of the ${work.year} ${work.kind} by ${authorNames}.`}
        image={work.seoImage.childImageSharp.resize.src}
        article
      />
      <main id="top" className={containerCss}>
        <header className={headerContainerCss}>
          <div className={headerTitleCss}>{work.title}</div>
          <div className={headerAuthorCss}>
            By{" "}
            {work.authors.map((author) => {
              return (
                <Link
                  className={authorLinkCss}
                  to={`/shelf/authors/${author.slug}`}
                  key={author.slug}
                >
                  {author.name}
                </Link>
              );
            })}
          </div>
          <div className={headerKindCss}>{work.kind}</div>
          <div className={headerYearCss}>First published in {work.year}.</div>
          {work.cover && (
            <GatsbyImage
              image={work.cover.childImageSharp.gatsbyImageData}
              alt={`A still from ${work.title} by ${toSentenceArray(
                work.authors.map((a) => a.name)
              ).join("")} (${work.year})`}
              loading={"eager"}
              className={coverCss}
            />
          )}
        </header>
        <ul className={reviewsListCss}>
          {work.reviews.map((review) => (
            <li
              className={reviewsListItemCss}
              key={review.frontmatter.sequence}
            >
              <article className={reviewCss}>
                <header
                  className={slugContainerCss}
                  id={review.frontmatter.sequence.toString()}
                >
                  <DateIcon className={reviewDateIconCss} />{" "}
                  <Grade
                    grade={review.frontmatter.grade}
                    className={reviewGradeCss}
                  />
                  <div>
                    <span className={slugOnCss}> on </span>
                    <span className={slugDateCss}>{review.dateFinished}</span>
                  </div>
                </header>
                <RenderedMarkdown
                  className={reviewContentCss}
                  // eslint-disable-next-line react/no-danger
                  text={review.linkedHtml}
                />
                <aside id="credits" className={reviewMetaCss}>
                  <div>
                    <dl>
                      <dt className={termCss}>Edition</dt>
                      <dd>
                        {review.frontmatter.edition} (
                        {review.frontmatter.editionNotes})
                      </dd>
                      <dt className={termCss}>Reading Time</dt>
                      <dd className={readingTimeContainerCss}>
                        {review.readingTime}{" "}
                        {review.readingTime === 1 ? "Day" : "Days"}
                        <div className={progressContainerCss}>
                          <div>
                            {review.frontmatter.progress[0].date}
                            {" – "}
                            <span className={progressMilestoneCss}>
                              Started
                            </span>
                          </div>
                          {review.frontmatter.progress.map((progress) => {
                            return (
                              <div key={progress.date}>
                                {progress.date}
                                {" – "}
                                {progress.percent === 100 ? (
                                  <span className={progressMilestoneCss}>
                                    Finished
                                  </span>
                                ) : (
                                  `${progress.percent}%`
                                )}
                              </div>
                            );
                          })}
                        </div>
                      </dd>
                    </dl>
                  </div>
                </aside>
              </article>
            </li>
          ))}
        </ul>
      </main>
      {structuredData && (
        <script
          type="application/ld+json"
          // eslint-disable-next-line react/no-danger
          dangerouslySetInnerHTML={{ __html: JSON.stringify(structuredData) }}
        />
      )}
    </Layout>
  );
}

interface PageQueryResult {
  work: Work;
}

export interface Work {
  title: string;
  year: number;
  kind: string;
  lastReviewGrade: string;
  authors: {
    name: string;
    slug: string;
    notes: string | null;
  }[];
  cover: {
    childImageSharp: {
      gatsbyImageData: IGatsbyImageData;
    };
  };
  seoImage: {
    childImageSharp: {
      resize: {
        src: string;
      };
    };
  };
  reviews: {
    frontmatter: {
      grade: string;
      edition: string;
      editionNotes: string | null;
      sequence: number;
      progress: {
        date: string;
        percent: number;
      }[];
    };
    readingTime: number;
    linkedHtml: string;
    dateFinished: string;
    dateFinishedIso: string;
  }[];
}

export const pageQuery = graphql`
  query ($slug: String!) {
    work: worksJson(slug: { eq: $slug }) {
      title
      year
      kind
      lastReviewGrade
      authors {
        name
        notes
        slug
      }
      reviews {
        frontmatter {
          grade
          sequence
          edition
          editionNotes: edition_notes
          progress {
            date(formatString: "DD MMM, YYYY")
            percent
          }
        }
        readingTime
        dateFinished(formatString: "DD MMM, YYYY")
        dateFinishedIso: dateFinished(formatString: "Y-MM-DD")
        linkedHtml
      }
      cover {
        childImageSharp {
          gatsbyImageData(
            layout: FIXED
            formats: [JPG, AVIF]
            quality: 80
            width: 250
            placeholder: TRACED_SVG
          )
        }
      }
      seoImage: cover {
        childImageSharp {
          resize(toFormat: JPG, width: 1200, quality: 80) {
            src
          }
        }
      }
    }
  }
`;
