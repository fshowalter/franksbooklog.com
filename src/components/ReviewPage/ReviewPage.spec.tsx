// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from "@testing-library/react";
import ReviewPage, { Head } from "./ReviewPage";
import data, { abandonedBook, audioBook } from "./ReviewPage.fixtures";

describe("/reviews/{slug}", () => {
  it("renders", () => {
    const { asFragment } = render(<ReviewPage data={data} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders for audio book", () => {
    const { asFragment } = render(<ReviewPage data={audioBook} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders for abandoned book", () => {
    const { asFragment } = render(<ReviewPage data={abandonedBook} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("sets page title", () => {
    render(<Head data={data} />);

    expect(document.title).toStrictEqual("On Writing by Stephen King");
  });
});
