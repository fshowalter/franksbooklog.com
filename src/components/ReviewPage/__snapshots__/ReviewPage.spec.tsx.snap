// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/reviews/{slug} renders 1`] = `
<DocumentFragment>
  <a
    class="skipLinkCss"
    href="#content"
  >
    Skip to content
  </a>
  <div
    class="containerCss"
  >
    <header
      class="mastHeaderCss"
    >
      <h1
        class="mastTitleCss"
      >
        <a
          href="/"
        >
          Frank's Book Log
        </a>
      </h1>
      <p
        class="mastTaglineCss"
      >
        Literature is a relative term.
      </p>
      <form
        accept-charset="UTF-8"
        action="https://www.google.com/search"
        class="mastSearchFormCss"
        method="get"
        role="search"
      >
        <label
          class="mastSearchWrapCss"
          for="search"
        >
          <div
            class="srOnlyCss"
          >
            Search
          </div>
          <input
            class="mastSearchInputCss"
            id="search"
            name="q"
            placeholder="Search..."
            type="text"
          />
          <input
            name="q"
            type="hidden"
            value="site:www.franksbooklog.com"
          />
          <button
            aria-label="Search"
            class="mastSearchSubmitCss"
            type="submit"
            value="Search"
          >
            <svg
              class="iconCss mastSearchIconCss"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10.442 10.442a1 1 0 011.415 0l3.85 3.85a1 1 0 01-1.414 1.415l-3.85-3.85a1 1 0 010-1.415z"
                fill-rule="evenodd"
              />
              <path
                d="M6.5 12a5.5 5.5 0 100-11 5.5 5.5 0 000 11zM13 6.5a6.5 6.5 0 11-13 0 6.5 6.5 0 0113 0z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </label>
      </form>
      <nav
        class="mastNavCss"
      >
        <ul
          class="mastNavListCss"
        >
          <li
            class="mastNavListItemCss"
          >
            <a
              class="mastNavLinkCss"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            class="mastNavListItemCss"
          >
            <a
              class="mastNavLinkCss"
              href="/about/"
            >
              About
            </a>
          </li>
          <li
            class="mastNavListItemCss"
          >
            <a
              class="mastNavLinkCss"
              href="/how-i-grade/"
            >
              How I Grade
            </a>
          </li>
          <li
            class="mastNavListItemCss"
          >
            <a
              class="mastNavLinkCss"
              href="/reviews/"
            >
              Reviews
            </a>
          </li>
          <li
            class="mastNavListItemCss"
          >
            <a
              class="mastNavLinkCss"
              href="/shelf/"
            >
              Shelf
            </a>
          </li>
        </ul>
      </nav>
    </header>
    <div
      class="childrenCss"
      id="content"
    >
      <main
        class="containerCss"
        id="top"
      >
        <header
          class="headerContainerCss"
        >
          <div
            class="headerTitleCss"
          >
            On Writing
          </div>
          <div
            class="headerAuthorCss"
          >
            By 
            <a
              class="authorLinkCss"
              href="/shelf/authors/undefined"
            >
              Stephen King
            </a>
          </div>
          <div
            class="headerKindCss"
          >
            Nonfiction
          </div>
          <div
            class="headerYearCss"
          >
            First published in 2000.
          </div>
          <div
            class="gatsby-image-wrapper coverCss"
            data-gatsby-image-wrapper=""
            style="position: relative; overflow: hidden; width: 250px; height: 400px;"
          >
            <img
              alt=""
              aria-hidden="true"
              data-placeholder-image=""
              decoding="async"
              src="data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20width='160'%20height='256'%20viewBox='0%200%20160%20256'%20preserveAspectRatio='none'%3e%3cpath%20d='M0%20128v128h161V0H0v128m35-55c-5%202-5%209-1%2012%203%203%206%203%209%200%205-6-1-15-8-12M6%20122H4v54h4c3%200%204%200%203-1l1-3c2-1%202-1%203%202%200%203%200%203%202%201l4-3c3-1%201-2-9-3-2%200-2%200%200-1%204-1%204-3%200-3-3%200-3%200-2-2s4-3%206-1c0%202%203%201%203%200l2-2v-1h-4l2-1%201-1-1-1v-3l2-18-3-3c-2%200-4-6-3-8s-4-3-6-3l-3%201m35%200l-13%201v3l-1%202-1-3-1-3-1%204%201%206%201%205c0%202%200%202-1%201-1-2-1-2-1%201%200%202%200%202-1%201h-1l2%203%202%203v-6l1%202c0%202%203%205%204%205%202-1%203%200%205%203%201%202%203%204%204%203v2c-1%203%205%209%207%207l3-1c2%200%202-1%200-4v-2l2%201h3l2-1c1-2%205-3%204-1v1h3l13-1c6%200%209%201%209%202l6%201h4l-4%204-6%204-1%201-3-1-3-4-2-2-5%202h-2v2c1%202%201%202-1%202l-2-1-2-1-7-1-4%201c1%201%200%201-1%201l2%201h4c1-1%207%200%207%201s4%202%205%201%202-1%203%201c4%203%2014%209%2015%208l4%202%203%202-4-1c-3%200-4%200-7%203-4%203-4%203-8%202-7-2-8-1-5%202%202%203%202%203%203%202h3c12%208%2020%2011%2023%207%201-1%201-1-1-1-1%200-2%200-3-2l-6-3-6-2c-2-2-2-2%201-4s5-2%207-2h2c0-1%201-1%205%201s6%203%206%201h1l6-4%205-5-1-5c0-6-3-12-5-12l-6%201c-5%202-5%202-5%200l1-3%201-2%201-1c2-1%202-1%201-3-2-4-6-3-6%201l-1%202-1-5c0-6%200-6%2010-8%206-1%209%206%206%2014-1%202-1%202%201%202%201%200%202-1%202-6l1-6c1-1%205%205%205%209l3%206c2%202%203%206%201%206l-2%201-2%203c-2%202-2%202%200%202h5c2%201%205%200%203-2l2-3h2l2%201c3%200%205%203%206%209%200%205%201%207%203%207l1-2c0-8%201-13%203-12h1v-12l1-4%201-6c1-1%202-4%201-6v-4l-5-1c-7%200-8-1-6-8%201-6%200-7-2-2l-2%204%201-4v-3h-18a3521%203521%200%2001-80%200m21%2019l-1%206c0%202%200%202%206%202l16%202c-2%200-2%200-1%201%202%202%203%200%202-7%200-6-1-7-2-6h-1l1-1c1-1-3-1-9-1H63l-1%204m92%2036c-2%2012-2%2012-5%2010-2-2-3-2-6-1l-10%201c-8%200-8%201-3%204l3%204c0%202%206%2010%208%2011l6%204c2%203%204%204%204%203l-4-5c-4-3-4-4-3-5%203-2%202-2-3-2l-3-1%204-1c5%200%206-3%202-5-2%200-2%200%200-1%203%200%204%200%206%203%203%203%206%204%206%201l-3-3c-4%200-5-2-2-3%202%200%203-1%202-2%200-3%201-2%202%200s1-1%201-10v-12l-2%2010m-29-2l-1%201%201%203v2l-1-1h-6l1%203v2c-1-1-2%200-2%202-1%204%200%205%202%201%201-3%202-3%204-3%201%201%202%200%204-1%200-2%202-3%203-3%202%200%202%200%200-1l-2-1h2l7-1%204-1-8-2h-8m-76%201v10h9c10%200%2011-1%209-3-1-1-2-2-1-4v-2l-17-1m82%2024l1%202c2%201%201%205%200%204l-1%201%2021%2025%204%205v-12l-12-13c-7-8-13-13-13-12m-42%2037l-74%204-1%203-1%202h60c60%200%2060%200%2059-2v-4c2-4%202-4-43-3'%20fill='%23d3d3d3'%20fill-rule='evenodd'/%3e%3c/svg%3e"
              style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
            />
            <picture>
              <source
                sizes="250px"
                srcset="/static/7008abb3d42ce4dd5b1b206c6503d358/1401a/on-writing-by-stephen-king.avif 250w,/static/7008abb3d42ce4dd5b1b206c6503d358/9bc36/on-writing-by-stephen-king.avif 500w"
                type="image/avif"
              />
              <img
                alt="A still from On Writing by Stephen King (2000)"
                data-gatsby-image-ssr=""
                data-main-image=""
                decoding="async"
                loading="eager"
                sizes="250px"
                src="/static/7008abb3d42ce4dd5b1b206c6503d358/ddf2c/on-writing-by-stephen-king.jpg"
                srcset="/static/7008abb3d42ce4dd5b1b206c6503d358/ddf2c/on-writing-by-stephen-king.jpg 250w,/static/7008abb3d42ce4dd5b1b206c6503d358/ef21d/on-writing-by-stephen-king.jpg 500w"
                style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
              />
            </picture>
            <noscript />
            <script
              type="module"
            >
              const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1)}}
            </script>
          </div>
        </header>
        <ul
          class="reviewsListCss"
        >
          <li
            class="reviewsListItemCss"
          >
            <article
              class="reviewCss"
            >
              <header
                class="slugContainerCss"
                id="4"
              >
                <svg
                  class="iconCss iconCss reviewDateIconCss"
                  height="1em"
                  viewBox="0 0 16 16"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title>
                    Watched on
                  </title>
                  <path
                    d="M14 2H2a1 1 0 0 0-1 1v11a1 1 0 0 0 1 1h12a1 1 0 0 0 1-1V3a1 1 0 0 0-1-1zM2 1a2 2 0 0 0-2 2v11a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V3a2 2 0 0 0-2-2H2z"
                    fill-rule="evenodd"
                  />
                  <path
                    d="M14 2H2a1 1 0 0 0-1 1v1h14V3a1 1 0 0 0-1-1zM2 1a2 2 0 0 0-2 2v2h16V3a2 2 0 0 0-2-2H2z"
                    fill-rule="evenodd"
                  />
                  <path
                    d="M3.5 0a.5.5 0 0 1 .5.5V1a.5.5 0 0 1-1 0V.5a.5.5 0 0 1 .5-.5zm9 0a.5.5 0 0 1 .5.5V1a.5.5 0 0 1-1 0V.5a.5.5 0 0 1 .5-.5z"
                    fill-rule="evenodd"
                  />
                </svg>
                 
                <img
                  alt="A-: 5 stars (out of 5)"
                  class="reviewGradeCss"
                  src="/svg/5-stars.svg"
                />
                <div>
                  <span
                    class="slugOnCss"
                  >
                     on 
                  </span>
                  <span
                    class="slugDateCss"
                  >
                    21 Mar, 2022
                  </span>
                </div>
              </header>
              <div
                class="renderedMarkdownCss reviewContentCss"
              >
                <p>
                  It’s hard to read 
                  <em>
                    The Shining
                  </em>
                   and not see Jack Nicholson. That’s how popular Stanley Kubrick’s film adaptation has become. New readers will likely be surprised that the Jack in the novel never utters the famous “Here’s Johnny!” line.
                </p>
                

                <!-- end -->
                

                <p>
                  The movie and book both share a similar plot. Jack Torrance takes a job as winter caretaker for the Overlook Hotel. He moves in with his wife and five-year-old son just as the last of the hotel staff are leaving for the off-season. Soon, the snow comes, blocking the roads and isolating the hotel. Cut off from the outside world, Jack goes crazy and turns against his family in a murderous rage.
                </p>
                

                <p>
                  Where the book and film differ is in their presentation of the supernatural. King makes it clear that a malevolent presence haunts the hotel. This evil force wants Jack’s son Danny for it’s own ends. King never makes it clear why, but it’s something to do with Danny’s burgeoning telepathy. Kubrick’s film invites us to consider that the supernatural elements are all in the character’s minds.
                </p>
                

                <p>
                  They’re both entertaining stories, but Kubrick’s version is the more insightful. Because Kubrick saw what it took King years to see; that in Jack Torrance, Stephen King was writing himself.
                </p>
                

                <p>
                  Consider the character of Jack Torrance. Jack is a broke, alcoholic playwright struggling to live up to his early successes and provide for his family.
                </p>
                

                <p>
                  Now consider Stephen King when he was writing 
                  <em>
                    The Shining
                  </em>
                  . King was in the throes of alcoholism. His first two novels had been wildly successful. But it wasn’t that long ago that King worked as a high school teacher and struggled to afford medicine for his infant child. Sure, there was money in the bank now, but what if none of his other books sold? What if, like Jack, he stopped being able to write?
                </p>
                

                <p>
                  King writes Jack as a victim of the hotel’s evil influence. Kubrick sees Jack as a monster and the story’s supernatural elements as a way for Jack to deflect responsibility. Knowing this, is it any wonder that King hated Kubrick’s version?
                </p>
              </div>
              <aside
                class="reviewMetaCss"
                id="credits"
              >
                <div>
                  <dl>
                    <dt
                      class="termCss"
                    >
                      Edition
                    </dt>
                    <dd>
                      Kindle ()
                    </dd>
                    <dt
                      class="termCss"
                    >
                      Reading Time
                    </dt>
                    <dd
                      class="readingTimeContainerCss"
                    >
                      3 Days
                      <div
                        class="progressContainerCss"
                      >
                        <div>
                          19 Mar, 2022 – 
                          <span
                            class="progressMilestoneCss"
                          >
                            Started
                          </span>
                        </div>
                        <div>
                          19 Mar, 2022 – 20%
                        </div>
                        <div>
                          20 Mar, 2022 – 71%
                        </div>
                        <div>
                          21 Mar, 2022 – 
                          <span
                            class="progressMilestoneCss"
                          >
                            Finished
                          </span>
                        </div>
                      </div>
                    </dd>
                  </dl>
                </div>
              </aside>
            </article>
          </li>
        </ul>
      </main>
      <script
        type="application/ld+json"
      >
        {"@context":"http://schema.org","@type":"AggregateRating","itemReviewed":{"@type":"Book","name":"On Writing","image":"/static/7008abb3d42ce4dd5b1b206c6503d358/8c3c2/on-writing-by-stephen-king.jpg"},"reviewCount":1,"ratingValue":5}
      </script>
    </div>
    <footer
      class="footerCss"
    >
      <ul
        class="footerNavListCss"
      >
        <li
          class="footerNavListItemCss"
        >
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="footerNavListItemCss"
        >
          <a
            href="/about/"
          >
            About
          </a>
        </li>
        <li
          class="footerNavListItemCss"
        >
          <a
            href="/how-i-grade/"
          >
            How I Grade
          </a>
        </li>
        <li
          class="footerNavListItemCss"
        >
          <a
            href="/reviews/"
          >
            Reviews
          </a>
        </li>
        <li
          class="footerNavListItemCss"
        >
          <a
            href="/shelf/"
          >
            Shelf
          </a>
        </li>
      </ul>
      <p
        class="footerFairUseCss"
      >
        All covers used in accordance with the 
        <a
          href="http://www.copyright.gov/title17/92chap1.html#107"
        >
          Fair Use Law.
        </a>
      </p>
      <div
        class="srOnlyCss"
      >
        <a
          href="#top"
        >
          To the top ↑
        </a>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
